.github/workflows/go.yml

```diff
@@ -23,11 +23,19 @@ jobs:
 
     - name: Test
       run: go test -v -coverprofile cover.out ./...
-      
+
     - name: Generate coverage report
       run: go tool cover -html=cover.out -o coverage.html
-      
+
     - uses: actions/upload-artifact@v2
       with:
         name: coverage
         path: ./coverage.html
+
+    - name: Check formatting
+      run: gofmt -s -l $(go list -f {{.Dir}} ./...)
+      continue-on-error: true
+
+    - name: Check old APIs
+      run: go tool fix -diff $(go list -f {{.Dir}} ./...)
```

Comment:
maybe we should consider adding [pre-commit](https://pre-commit.com/) ?
im on phone rn so cant rly check exit code of go tool fix -diff, does it give 0 on no changes and other exit code when it prints the changes?

Reply:
I think having this check done this way in CI is ok for now. fix -diff should return 0. Still these check have continue-on-error as I don't want to fail everything due to style check, but still such information is valuable.

These steps may be removed in the future when other CI tools are added (e.g. golangci-lint).